addon/initializers/m3-store.js

```diff
@@ -53,6 +88,36 @@ const STORE_OVERRIDES = {
     return this._super(modelName);
   },
 
+  instantiateRecord(identifier, createRecordArgs, recordDataFor, notificationManager) {
+    let recordData = recordDataFor(identifier);
+    let createOptions = createRecordArgs;
+    recordDataToQueryCache.set(recordData, this._queryCache);
+    let modelName = identifier.type;
+    // TODO NOW deal with this
+    if (get(this, '_schemaManager').includesModel(modelName)) {
+      delete createOptions.container;
+      delete createOptions.currentState;
+      delete createOptions._internalModel;
+      createOptions._recordData = recordData;
+      createOptions.store = this;
```

Comment:
I very much doubt that we actually have to mutate user-supplied args any longer now that we are instantiating the record ourselves; let's not mutate `createOptions`